frontend/src/components/maps/leaflet/InfoSlideOut/HeaderActions.tsx

```diff
@@ -0,0 +1,122 @@
+import styled from 'styled-components';
+import * as React from 'react';
+import { Row } from 'react-bootstrap';
+import { IBuilding, IParcel, PropertyTypes } from 'actions/parcelsActions';
+import { useLeaflet } from 'react-leaflet';
+import { MAX_ZOOM } from 'constants/strings';
+import { Link, useLocation } from 'react-router-dom';
+import queryString from 'query-string';
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import Claims from 'constants/claims';
+
+const LinkMenu = styled(Row)`
+  background-color: #f2f2f2;
+  height: 35px;
+  width: 341px;
+  margin-left: -10px;
+  font-size: 14px;
+  font-family: ProximaNova, sans-serif;
+  padding: 10px;
+  a {
+    padding: 0 10px;
+    color: #1a5a96;
+  }
+`;
+
+const VerticalBar = styled.div`
+  border-left: 6px solid rgba(96, 96, 96, 0.2);
+  height: 18px;
+  margin-left: 1%;
+  margin-right: 1%;
+  border-width: 2px;
+`;
+
+export type IHeaderActions = {
+  /** The selected property */
+  propertyInfo: IParcel | IBuilding | null;
+  /** the selected property type */
+  propertyTypeId: PropertyTypes | null;
+  /** additional action to be taken when a link in the menu is clicked */
+  onLinkClick?: () => void;
+};
+
+/**
+ * Actions that can be done on the property info slide out
+ * @param propertyInfo the selected property information
+ * @param propertyTypeId the selected property type
+ * @param onLinkClick additional action on menu item click
+ */
+const HeaderActions: React.FC<IHeaderActions> = ({ propertyInfo, propertyTypeId, onLinkClick }) => {
+  const leaflet = useLeaflet();
+  const location = useLocation();
+  const keycloak = useKeycloakWrapper();
+  const curZoom = leaflet.map?.getZoom();
+  const jumpToView = () =>
+    leaflet.map?.setView(
+      [propertyInfo?.latitude as number, propertyInfo?.longitude as number],
+      MAX_ZOOM,
+    );
+  const zoomToView = () =>
+    leaflet.map?.flyTo(
+      [propertyInfo?.latitude as number, propertyInfo?.longitude as number],
+      MAX_ZOOM,
+    );
+
+  const buildingId = propertyTypeId === PropertyTypes.BUILDING ? propertyInfo?.id : undefined;
+  const parcelId = propertyTypeId === PropertyTypes.PARCEL ? propertyInfo?.id : undefined;
+  return (
+    <LinkMenu>
+      Actions:
+      <Link
+        onClick={() => {
+          if (curZoom! < MAX_ZOOM) jumpToView();
```

Comment:
This is probably fine, but I generally think about using ! in the sense of what does this get me?
Well, generally curZoom should exist as leaflet.map?.getZoom() should generally be available. However in the off chance that is is null or undefined curZoom! will blow up and leave a nasty error.

using some sort of sensible default would prevent the nasty error at least. Alternatively you could just not cull jumpToView if curZoom is not known

Reply:
Also, does this if make sense? It seems like if we are zoomed in more then max zoom we would still want to jump to the property right? we may not want to change the zoom level in that case, but moving the center of the map seems to make sense.

Reply:
I hear ya, that makes sense 👍